module/VuFind/src/VuFind/Db/Service/AccessTokenService.php

```diff
@@ -131,10 +156,11 @@ public function getNonce(int $userId): ?string
     public function deleteExpired(DateTime $dateLimit, ?int $limit = null): int
     {
         $subQueryBuilder = $this->entityManager->createQueryBuilder();
+        // Delete only tokens with expires set to 1
         $subQueryBuilder->select('CONCAT(a.id, a.type)')
             ->from(AccessTokenEntityInterface::class, 'a')
-            ->where('a.created < :latestCreated')
-            ->setParameter('latestCreated', $dateLimit->format(VUFIND_DATABASE_DATETIME_FORMAT));
+            ->where('a.created < :latestCreated AND a.expires = 1')
+            ->setParameter('latestCreated', $dateLimit->format('Y-m-d H:i:s'));
```

Comment:
I think you probably want to reintroduce the constant here:

```suggestion
            ->setParameter('latestCreated', $dateLimit->format(VUFIND_DATABASE_DATETIME_FORMAT));
```